static/app/components/assigneeSelectorDropdown.tsx

```diff
@@ -0,0 +1,543 @@
+      label: t('Teams'),
+      options: assignableTeamList?.map(makeTeamOption) ?? [],
+    };
+
+    const suggestedUsers = suggestedAssignees?.filter(
+      assignee => assignee.type === 'user'
+    );
+    const suggestedTeams = suggestedAssignees?.filter(
+      assignee => assignee.type === 'team'
+    );
+
+    const suggestedOptions = {
+      value: '_suggested_assignees',
+      label: t('Suggested Assignees'),
+      options:
+        suggestedUsers
+          .map(makeSuggestedAssigneeOption)
+          .concat(suggestedTeams?.map(makeSuggestedAssigneeOption)) ?? [],
+    };
+
+    options.push(suggestedOptions, memberOptions, teamOptions);
+
+    return options;
+  };
+
+  const makeTrigger = (props, isOpen) => {
+    const avatarElement = (
+      <AssigneeAvatar
+        assignedTo={group.assignedTo}
+        suggestedActors={getSuggestedAssignees()}
+      />
+    );
+    return (
+      <Fragment>
+        {loading && (
+          <LoadingIndicator mini style={{height: '24px', margin: 0, marginRight: 11}} />
+        )}
+        {!loading && !noDropdown && (
+          <DropdownButton data-test-id="assignee-selector" {...props}>
+            {avatarElement}
+            <Chevron direction={isOpen ? 'up' : 'down'} size="small" />
+          </DropdownButton>
+        )}
+        {!loading && noDropdown && avatarElement}
+      </Fragment>
+    );
+  };
+
+  const makeFooterInviteButton = () => {
+    return (
+      <Button
+        size="xs"
+        aria-label={t('Invite Member')}
+        disabled={loading}
+        onClick={(event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+          event.preventDefault();
+          openInviteMembersModal({source: 'assignee_selector'});
+        }}
+        icon={<IconAdd isCircled />}
+      >
+        {t('Invite Member')}
+      </Button>
+    );
+  };
+
+  return (
+    <AssigneeWrapper>
+      <CompactSelect
+        searchable
+        clearable
+        disallowEmptySelection={false}
+        onClick={e => e.stopPropagation()}
+        value={
+          group.assignedTo
+            ? `${group.assignedTo?.type === 'user' ? 'user:' : 'team:'}${group.assignedTo.id}`
+            : ''
+        }
+        onClear={() => handleSelect(null)}
+        menuTitle={t('Select Assignee')}
+        searchPlaceholder="Search users or teams..."
+        size="sm"
+        onChange={handleSelect}
+        options={makeAllOptions()}
+        trigger={makeTrigger}
+        menuFooter={makeFooterInviteButton()}
+      />
+    </AssigneeWrapper>
+  );
+}
+
+const AssigneeWrapper = styled('div')`
+  display: flex;
+  justify-content: flex-end;
+
+  /* manually align menu underneath dropdown caret */
+`;
+
+const DropdownButton = styled('button')`
+  appearance: none;
+  border: 0;
+  background: transparent;
+  display: flex;
+  align-items: center;
+  font-size: 20px;
+  gap: ${space(0.5)};
+`;
+
+const StyledIconUser = styled(IconUser)`
+  /* We need this to center with Avatar */
```

Comment:
just put this two lines higher outside the template tag, it has trouble removing these in production